migration/migrator/migrations/course/20240209021641_lecture_chat.py

```diff
@@ -0,0 +1,48 @@
+"""Migration for a given Submitty course database."""
+
+import json
+from pathlib import Path
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute("CREATE TABLE IF NOT EXISTS chatrooms(id SERIAL PRIMARY KEY, host_id character varying NOT NULL, host_name character varying, title text NOT NULL, description text, is_active BOOLEAN DEFAULT false NOT NULL, allow_anon BOOLEAN DEFAULT true NOT NULL)")
+    database.execute("CREATE TABLE IF NOT EXISTS chatroom_messages(id SERIAL PRIMARY KEY, chatroom_id integer NOT NULL, userId character varying NOT NULL, display_name character varying, role character varying, content text NOT NULL, timestamp timestamp(0) with time zone NOT NULL)")
```

Comment:
These need foreign keys to be set up wherever possible.  Chatroom messages should have foreign-key relationships with chatrooms and users at a minimum.

As an aside: what is the purpose of the `host_id` and `host_name` columns?

Reply:
These columns currently don't have other purposes except for showing the name of the host on students' UI. I intended. to include them to prepare for if we plan to have different hosts hosting different chat rooms in the future.